packages/server/api/src/app/core/security/security-handler.ts

```diff
@@ -14,3 +14,14 @@ export abstract class BaseSecurityHandler implements SecurityHandler {
     protected abstract canHandle(request: FastifyRequest): Promise<boolean>
     protected abstract doHandle(request: FastifyRequest): Promise<void>
 }
+
+export abstract class BaseAuthzHandler<TRequest extends FastifyRequest> implements SecurityHandler {
```

Comment:
I think this is overkill, we don't need this abstraction and above abstraction at all.